### ISSUE TITLE
Need EncodedChunk to have video metadata to support WebRTC

### ISSUE THREAD

**Issue by alvestrand** [w3c/webcodecs#245]
The WebRTC Encoded Transform spec's VideoFrame object (https://w3c.github.io/webrtc-encoded-transform/#rtcencodedvideoframe) has a getMetadata function that returns information about the frame that is produced by the encoder, and needs to be passed on to consumers (and vice versa for decoders).

If we're going to harmonize RTCEncodedVideoFrame with EncodedVideoChunk, we need some way to provide this functionality.

(This bug is created as part of bug triage on webrtc-encoded-transform; all bugs that are about buffer formats should move here, I think.)

**Comment by aboba** [comment]
MetaData Proposal: https://docs.google.com/document/d/1O6kxAQPLWhCRHkH-6zutcqEUOUWRS6NFM2CQAvi1VTg/

**Comment by aboba** [comment]
@alvestrand Looks like we need to plan for metadata extensibility, so moving the SVC metadata into its own `svc` dictionary probably makes sense.

**Comment by aboba** [comment]
@alvestrand @Djuffin @padenot   Do we still believe that changes to EncodedChunk are needed to support WebRTC?  Or is the required metadata being added to RTCEncoded{Audio, Video}Frame instead?

**Comment by padenot** [comment]
For WebRTC folks to answer.

**Comment by aboba** [comment]
@alvestrand Is this still an issue, or can we close this?

**Comment by alvestrand** [comment]
The metadata that matters is the stuff that is needed to create the RTCEncoded{Audio, Video}Frame metadata. Offhand, I'm thinking of the dependency maps that go into the dependency descriptor, but there are probably more. I don't know at the moment how to find these (and have mislaid the document I was working on for it). But I don't think the issue can be closed.

**Comment by alvestrand** [comment]
Here's the table from my half-baked doc:

The RTCEencodedVideoFrameMetadata for an outgoing frame can be sourced as follows:

    unsigned long long frameId;
    sequence<unsigned long long> dependencies; - from encoding config
    unsigned short width; - from encoding config
    unsigned short height; - from encoding config
    unsigned long spatialIndex; - from encoding config
    unsigned long temporalIndex; - from encoding config
    unsigned long synchronizationSource; - irrelevant, added by sending process
    octet payloadType; - irrelevant, added by sending process
    sequence<unsigned long> contributingSources; - must be configured if used
    long long timestamp;    // microseconds - copied from EncodedVideoChunk
    unsigned long rtpTimestamp; - irrelevant, added by sending process
    DOMString mimeType; - from encoding config

The frameId is the one that sticks out as "missing". I think it's used by the dependency descriptor (which is *not* exposed as RTCEncodedVideoFrameMetadata, it seems).

**Comment by aboba** [comment]
Yes, `frame_number` (a 16-bit relative version of `frameId`) is used by the [Dependency Descriptor](https://aomediacodec.github.io/av1-rtp-spec/#appendix).

While in principle the dependency descriptor is derivable from the metadata cited plus the `scalabilityMode`, in practice this would be a *lot* of (duplicative) work to derive.

**Comment by Djuffin** [comment]
Media WG on Dec 10, 2024: 
[EncodedVideoChunkMetadata](https://www.w3.org/TR/webcodecs/#encoded-video-chunk-metadata) is only produced by VideoEncoder, it's not available for chunks coming from network and so one.

Let's just add constructors that allow developers to create  
[RTCEncodedVideoFrame](https://w3c.github.io/webrtc-encoded-transform/#rtcencodedvideoframe) and [EncodedVideoChunk](https://www.w3.org/TR/webcodecs/#encodedvideochunk-interface) from each other.
